### ISSUE TITLE
Status bar doesn't show intensity if layers are invisible

### ISSUE THREAD

**Issue by haesleinhuepf** [napari/napari#1872]
## 🐛 Bug
Hi all,

TL;DR: The intensity-display in the status bar behaves unintuitive IMHO. I'll send a potential bugfix as PR as well.

assume you have two layers. If you move your mouse in the image, you see mouse position and intensity in the status bar:
![image](https://user-images.githubusercontent.com/12660498/99156169-210c5f00-26bf-11eb-8336-e78637282868.png)

If you then turn visibility of the top layer off, you don't see the intensity anymore. Furthermore, it appears like multiple layers put information in the status bar. The text is blinking.
![image](https://user-images.githubusercontent.com/12660498/99156189-4b5e1c80-26bf-11eb-98ef-eb7297599e9e.png)

## To Reproduce

1. run this code
```
from skimage import data

import napari
with napari.gui_qt():
    viewer = napari.Viewer()
    viewer.add_image(data.astronaut())
    viewer.add_image(data.astronaut())
```
2. move the mouse in the image, read intensity in the status bar
3. turn the upper layer invisible
4. move the mouse again, intensity isn't shown; independent from which layer is selected

## Expected behavior

I would argue that either in general the intensity of the selected layer is shown. Alternatively the intensity of the upper-most visible layer is shown.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:
napari: 0.4.1a2.dev5+g20f3479
Platform: Windows-10-10.0.18362-SP0
Python: 3.7.7 (default, May 6 2020, 11:45:54) [MSC v.1916 64 bit (AMD64)]
Qt: 5.9.6
PyQt5: 5.9.2
NumPy: 1.18.5
SciPy: 1.5.0
Dask: 2.20.0
VisPy: 0.6.4

GL version: 4.6.13587 Compatibility Profile Context 19.40.38.11 26.20.14038.11003
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari_plugin_engine: 0.1.8
- svg: 0.1.4

 - Any other relevant information:

**Comment by haesleinhuepf** [comment]
To be discussed: I think it would also be cool if the layer-selection would be taken into account for displaying intensities. Furthermore, I think at the moment, multiple layers refresh the status bar, but the uppermost does it last. Maybe layers can take into account if they are selected?

**Comment by sofroniewn** [comment]
This may be related to #1836, which is a regression from the last release, so I'm not surprised to hear behaviour is off right now. I'll take a look at the fix
